### DISCUSSION TITLE
Tip: skipping IR drop analysis

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#826] [General] [answered]
It can be useful to skip analyze_power_grid, if the analysis of the default power grid is causing problems.

The trick is that an empty environment variable is still detected as set by https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/8307935f0f186cbfea4cc14dcbd17a0c9c8fafd1/flow/scripts/final_report.tcl#L39

To get around this, unset ```PWR_NETS_VOLTAGES``` and  ```GND_NETS_VOLTAGES``` after the default values are set in the platform config.mk file: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/8307935f0f186cbfea4cc14dcbd17a0c9c8fafd1/flow/platforms/asap7/config.mk#L255

This can be done by creating a private.mk file that is included at the end of the Makefile: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/8307935f0f186cbfea4cc14dcbd17a0c9c8fafd1/flow/Makefile#L803

Create a private.mk file:

```
# Skip analyze_power_grid
undefine PWR_NETS_VOLTAGES
undefine GND_NETS_VOLTAGES
```

In the config.mk for your design, point to the folder where the private.mk file is located:

```
export PRIVATE_DIR = ./designs/$(PLATFORM)/mydesigname/
```

**Comment by maliberty** [comment, marked as answer]
You could also change the check to also skip on an empty string if that's easier.

**Reply by oharboe** [reply]
I prefer not to change ORFS and these checks as stated in the .tcl scripts are everywhere in the code, so I wanted a trick that worked in all other cases
